### PR TITLE
Fixed documentation to work with project.toml instead of setup.py

### PR DESCRIPTION
**Pull request by susuhahnml**

### CODE REVIEW

**Comment by susuhahnml** [comment]
@rkaminsk SHould I also remove the case for python 3.7 in the code then? in the parser.py file

**Comment by rkaminsk** [comment]
> @rkaminsk SHould I also remove the case for python 3.7 in the code then? in the parser.py file

Yes, please do this. Python 3.9 is the latest release still supported. It is not worth the headache to support earlier versions. We could even drop python 3.9 and go for 3.10 if this makes our lives easier.

File: pyproject.toml [hunk 1]
```diff
@@ -1,31 +1,26 @@
...
-lint_pylint = [ "pylint" ]
-typecheck = [ "types-setuptools", "mypy" ]
-test = [ "coverage[toml]" ]
-doc = [ "sphinx", "furo", "nbsphinx", "sphinx_copybutton", "myst-parser" ]
-dev = [ "fillname[test,typecheck,lint_pylint]" ]
+format = ["black", "isort", "autoflake"]
+lint_pylint = ["pylint"]
+typecheck = ["types-setuptools", "mypy"]
+test = ["coverage[toml]"]
+doc = ["sphinx", "furo", "nbsphinx", "sphinx_copybutton", "myst-parser", "importlib_metadata", "toml"]
```
Comment (rkaminsk) [inline pyproject.toml:22]: The importlib_metadata dependency can also be dropped.

**Comment by susuhahnml** [comment]
Read to merge (I think) @rkaminsk

File: doc/conf.py [hunk 2]
```diff
@@ -18,23 +20,23 @@
 
 # -- Project information -----------------------------------------------------
 
-_config = configparser.RawConfigParser()
-_config.read(os.path.join("..", "setup.cfg"))
-_meta = dict(_config.items("metadata"))
+toml_config = toml.load(os.path.join("..", "pyproject.toml"))
+project = toml_config["project"]["name"]
```
Comment (rkaminsk) [inline doc/conf.py:24]: project dose not seem to be used

File: doc/conf.py [hunk 3]
```diff
@@ -18,23 +20,23 @@
...
+author = ", ".join([a["name"] for a in toml_config["project"]["authors"]])
+url = toml_config["project"]["urls"]["Homepage"]
+copyright = f'{datetime.datetime.now().date().strftime("%Y")}, {author}'
```
Comment (rkaminsk) [inline doc/conf.py:27]: does not seem to be used

File: doc/conf.py [hunk 4]
```diff
@@ -18,23 +20,23 @@
 
 # -- Project information -----------------------------------------------------
 
-_config = configparser.RawConfigParser()
-_config.read(os.path.join("..", "setup.cfg"))
-_meta = dict(_config.items("metadata"))
+toml_config = toml.load(os.path.join("..", "pyproject.toml"))
+project = toml_config["project"]["name"]
+author = ", ".join([a["name"] for a in toml_config["project"]["authors"]])
```
Comment (rkaminsk) [inline doc/conf.py:25]: author does not seem to be used

File: src/fillname/utils/parser.py [hunk 5]
```diff
@@ -2,20 +2,15 @@
 The command line parser for the project.
 """
 
-import sys
 from argparse import ArgumentParser
+from importlib import metadata  # nocoverage
```
Comment (rkaminsk) [inline src/fillname/utils/parser.py:6]: we can remove nocoverage now

File: doc/conf.py (hunk 4 above)
Comment (rkaminsk) [reply doc/conf.py:25]: It is actually used by sphinx: https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-author

File: doc/conf.py (hunk 3 above)
Comment (rkaminsk) [reply doc/conf.py:27]: It is actually used by sphinx: https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-copyright

**Pull request merged**